diff --git a/sites/all/modules/contrib/nodeformcols/nodeformcols.module b/sites/all/modules/contrib/nodeformcols/nodeformcols.module
index a215f37..f6400d7 100644
--- a/sites/all/modules/contrib/nodeformcols/nodeformcols.module
+++ b/sites/all/modules/contrib/nodeformcols/nodeformcols.module
@@ -156,8 +156,7 @@ function template_preprocess_node_form(&$aVars) {
   $adjust_to_buttons = $placements['buttons']['region'] == $default_region;
 
   foreach ($form as $key => $field) {
-    if (substr($key, 0, 1)=='#' ||
-        $field['#type']=='value' || $field['#type']=='hidden'|| $field['#type']=='token') {
+    if (substr($key, 0, 1) == '#' || (isset($field['#type']) && ($field['#type'] == 'value' || $field['#type'] == 'hidden'|| $field['#type'] == 'token'))) {
       continue;
     }
 
