diff --git a/sites/all/modules/contrib/oembed/oembedcore.admin.inc b/sites/all/modules/contrib/oembed/oembedcore.admin.inc
index 94b9bf1..d2659ce 100644
--- a/sites/all/modules/contrib/oembed/oembedcore.admin.inc
+++ b/sites/all/modules/contrib/oembed/oembedcore.admin.inc
@@ -633,7 +633,7 @@ function oembedcore_enable_provider($provider) {
   }
   ctools_include('export');
   ctools_export_set_status('oembedcore_provider', $provider->name, FALSE);
-  cache_clear_all('oembedcore:providers');
+  cache_clear_all('oembedcore:providers', 'cache');
   drupal_goto();
 }
 
@@ -646,6 +646,6 @@ function oembedcore_disable_provider($provider) {
   }
   ctools_include('export');
   ctools_export_set_status('oembedcore_provider', $provider->name, TRUE);
-  cache_clear_all('oembedcore:providers');
+  cache_clear_all('oembedcore:providers', 'cache');
   drupal_goto();
 }
\ No newline at end of file
diff --git a/sites/all/modules/contrib/oembed/oembedcore.module b/sites/all/modules/contrib/oembed/oembedcore.module
index eb46d11..abe21ad 100644
--- a/sites/all/modules/contrib/oembed/oembedcore.module
+++ b/sites/all/modules/contrib/oembed/oembedcore.module
@@ -546,7 +546,7 @@ function oembedcore_provider_load_all() {
 function oembedcore_provider_save(&$provider) {
   $update = (isset($provider->pid)) ? array('pid') : array();
   drupal_write_record('oembedcore_provider', $provider, $update);
-  cache_clear_all('oembedcore:providers');
+  cache_clear_all('oembedcore:providers', 'cache');
   return $preset;
 }
 
@@ -555,7 +555,7 @@ function oembedcore_provider_save(&$provider) {
  */
 function oembedcore_provider_delete($provider) {
   db_query("DELETE FROM {oembedcore_provider} WHERE name = '%s' AND pid = %d", $provider->name, $provider->pid);
-  cache_clear_all('oembedcore:providers');
+  cache_clear_all('oembedcore:providers', 'cache');
 }
 
 /**
